gcsfs/core.py

```diff
@@ -574,28 +581,149 @@ async def _list_objects(self, path, prefix="", versions=False):
             else:
                 return []
         out = pseudodirs + items
-        # Don't cache prefixed/partial listings
-        if not prefix:
+        
+        use_snapshot_listing = False if not inventory_report_info \
+            else inventory_report_info.get("use_snapshot_listing")
+
+        # Don't cache prefixed/partial listings, in addition to 
+        # not using the inventory report service to do listing directly.
+        if not prefix and use_snapshot_listing is False:
             self.dircache[path] = out
         return out
 
     async def _do_list_objects(
-        self, path, max_results=None, delimiter="/", prefix="", versions=False
-    ):
+        self, path, max_results=None, delimiter="/", 
+                    prefix="", versions=False, **kwargs):
+        
         """Object listing for the given {bucket}/{prefix}/ path."""
         bucket, _path, generation = self.split_path(path)
         _path = "" if not _path else _path.rstrip("/") + "/"
         prefix = f"{_path}{prefix}" or None
 
+        # Page size of 5000 is officially supported across GCS.
```

Comment:
nit: worth a reference here?

Reply:
> nit: worth a reference here?

I believe this is already GA'ed. Users can verify by themselves that the HTTP  response is indeed returning 5000 objects per page, rather than 1000 objects per page. However, I think the public documentation is slightly outdated and have not updated this yet.

Reply:
Maybe our own documentation (i.e., the prose) should have these facts and links to wherever they are mentioned, even if it's not official released user docs.

Reply:
Sounds good to me!